File: Lesson11-Completeing-The-UI/T11.02-Exercise-DataBinding/app/src/main/java/com/example/android/boardingpass/MainActivity.java
package com.example.android.boardingpass;

/*
* Copyright (C) 2016 The Android Open Source Project
*
* Licensed under the Apache License, Version 2.0 (the "License");
* you may not use this file except in compliance with the License.
* You may obtain a copy of the License at
*
*      http://www.apache.org/licenses/LICENSE-2.0
*
* Unless required by applicable law or agreed to in writing, software
* distributed under the License is distributed on an "AS IS" BASIS,
* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
* See the License for the specific language governing permissions and
* limitations under the License.
*/

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.example.android.boardingpass.databinding.ActivityMainBinding;
import com.example.android.boardingpass.utilities.FakeDataUtils;

import java.text.SimpleDateFormat;
import java.util.Locale;
import java.util.concurrent.TimeUnit;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        BoardingPassInfo boardingPassInfo = FakeDataUtils.generateFakeBoardingPassInfo();
        displayBoardingPassInfo(boardingPassInfo);
    }

    private void displayBoardingPassInfo(BoardingPassInfo info) {
        binding.textViewPassengerName.setText(info.passengerName);
        binding.textViewDestinationAirport.setText(info.destCode);
        binding.textViewOriginAirport.setText(info.originCode);
        binding.textViewGate.setText(info.departureGate);
        binding.textViewTerminal.setText(info.departureTerminal);
        binding.textViewSeat.setText(info.seatNumber);

        SimpleDateFormat formatter = new SimpleDateFormat(getString(R.string.timeFormat), Locale.getDefault());
        String boardingTime = formatter.format(info.boardingTime);
        String departureTime = formatter.format(info.departureTime);
        String arrivalTime = formatter.format(info.arrivalTime);

        binding.textViewBoardingTime.setText(boardingTime);
        binding.textViewDepartureTime.setText(departureTime);
        binding.textViewArrivalTime.setText(arrivalTime);

        long totalMinutesUntilBoarding = info.getMinutesUntilBoarding();
        long hoursUntilBoarding = TimeUnit.MINUTES.toHours(totalMinutesUntilBoarding);
        long minutesLessHoursUntilBoarding =
                totalMinutesUntilBoarding - TimeUnit.HOURS.toMinutes(hoursUntilBoarding);

        String hoursAndMinutesUntilBoarding = getString(R.string.countDownFormat,
                hoursUntilBoarding,
                minutesLessHoursUntilBoarding);

        binding.textViewBoardingInCountdown.setText(hoursAndMinutesUntilBoarding);
        binding.textViewTerminal.setText(info.departureTerminal);
        binding.textViewGate.setText(info.departureGate);
        binding.textViewSeat.setText(info.seatNumber);
    }
}

File: Lesson04a-Starting-New-Activities/T04a.01-Exercise-AddNewActivity/app/src/main/java/com/example/android/explicitintent/MainActivity.kt
/*
 * Copyright (C) 2016 The Android Open Source Project
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.example.android.explicitintent

import android.content.Context
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.view.View
import android.view.View.OnClickListener
import android.widget.Button
import android.widget.EditText
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    /* Fields that will store our EditText and Button */
    private var mNameEntry: EditText? = null
    private var mDoSomethingCoolButton: Button? = null

    // onCreate
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        /*
             * Using findViewById, we get a reference to our Button from xml. This allows us to
             * do things like set the onClickListener which determines what happens when the button
             * is clicked.
             */
        mDoSomethingCoolButton = findViewById(R.id.b_do_something_cool) as Button
        mNameEntry = findViewById(R.id.et_text_entry) as EditText

        /* Setting an OnClickListener allows us to do something when this button is clicked. */
        mDoSomethingCoolButton!!.setOnClickListener {
            val context = this@MainActivity
            val message = "Button clicked!\nTODO: Start a new Activity and pass some data."
            Toast.makeText(context, message, Toast.LENGTH_LONG).show()
        }
    }
}
